src/v/cluster/security_frontend.cc

```diff
@@ -308,4 +309,59 @@ security_frontend::dispatch_delete_acls_to_leader(
       });
 }
 
+/**
+ * For use during cluster creation, if RP_BOOTSTRAP_USER is set
+ * then write a user creation message to the controller log.
+ *
+ * @returns an error code if controller log write failed.  If the
+ *          environment variable is missing or malformed this is
+ *          not considered an error.
+ *
+ */
+ss::future<std::error_code> security_frontend::maybe_create_bootstrap_user() {
+    static const ss::sstring bootstrap_user_env_key{"RP_BOOTSTRAP_USER"};
+
+    auto creds_str_ptr = std::getenv(bootstrap_user_env_key.c_str());
+    if (creds_str_ptr == nullptr) {
+        // Environment variable is not set
+        co_return errc::success;
+    }
+
+    ss::sstring creds_str = creds_str_ptr;
+    auto colon = creds_str.find(":");
+    if (colon == ss::sstring::npos || colon == creds_str.size() - 1) {
+        // Malformed value.  Do not log the value, it may be malformed
+        // but it is still a secret.
+        vlog(
+          clusterlog.warn,
+          "Invalid value of {} (expected \"username:password\")",
+          bootstrap_user_env_key);
+        co_return errc::success;
```

Comment:
surprised that malformed input is reported as success rather than an error so that `cluster_creation_hook` will fail on startup to provide feedback to user that early user creation failed.

Reply:
nevermind. i see that the comment on the function explicitly lists malformed input as a success case.